program-runtime/src/loaded_programs.rs

```diff
@@ -941,26 +1012,30 @@ impl<FG: ForkGraph> LoadedPrograms<FG> {
         keys.iter().for_each(|key| self.unload_program(key));
     }
 
+    fn unload_program_entry(&mut self, program: &Pubkey, remove_entry: &Arc<LoadedProgram>) {
+        if let Some(entries) = self.entries.get_mut(program) {
+            if let Some(candidate) = entries.iter_mut().find(|entry| entry == &remove_entry) {
+                if let Some(unloaded) = candidate.to_unloaded() {
```

Comment:
any of these being `None` means (and _meant_ prior to this change) that we're failing silently

Reply:
I'll add some `error!` logs for these conditions.
Ideally it should never happen, since the entry was looked up before we tried to remove it.

Reply:
i imagine panicking is more appropriate given, as you said, if this happens, something is very wrong and it is unlikely that proceeding is safe